packages/babel-plugin-proposal-class-properties/test/fixtures/general/non-block-arrow-func/expected.js

```diff
@@ -1,27 +1,32 @@
 export default (param => {
-  var _class, _temp;
+  var _class;
 
-  return _temp = _class =
+  return _class =
```

Comment:
Is it possible to remove this unused `_class =` assignment?

Reply:
it probably is, added `_class` is caused by `ClassExpression` being replaced by multiple statements. The `path#replaceWithMultiple` creates an IIFE when the conversion is made so the single expression can be maintained in place.

This is certainly something I'd like to revisit, but for another month (till Jan 16th) I have very little time and I couldn't manage to do this properly at the moment because of the complexity and probably my insufficient experience with AST manipulation.

Ideally I would like to "inline" the `ClassExpression` here if possible to skip second IIFE creation, but I'm not sure how safe it is because scope references have to be synced and stuff.

Reply:
This case (and other similar) are caused by simple insertion [here](https://github.com/babel/babel/blob/f3410e761ceff73cf7c2c45bb79fff6479a28d5b/packages/babel-plugin-proposal-class-properties/src/index.js#L228-L230). To remove this unused variable the logic here would have to be expanded. I might revisit this in a followup PR, but would like to keep this one as simple as possible.